browser/lib/markdown-it-frontmatter.js

```diff
@@ -0,0 +1,21 @@
+'use strict'
+
+module.exports = function frontMatterPlugin (md) {
+  function frontmatter (state, startLine, endLine, silent) {
+    if (startLine !== 0 || state.src.substr(startLine, state.eMarks[0]) !== '---') {
+      return false
+    }
+
+    let line = 0
+    while (++line < state.lineMax && state.src.substring(state.bMarks[line], state.eMarks[line]) !== '---') {
```

Comment:
I still think we should display the markdown if the front-matter syntax is invalid. For example:
```
---
layout: default
-----
sdasdasdasdasd
```
Should display:
![image](https://user-images.githubusercontent.com/12984316/45260564-184fdc80-b415-11e8-99b3-87cd68e11295.png)
With the current behavior (hide all content) the user will get confuse.

Here is my suggestion for the plugin:
```js
module.exports = function frontMatterPlugin (md) {
  function frontmatter (state, startLine, endLine, silent) {
    if (startLine !== 0 || state.src.substr(startLine, state.eMarks[0]) !== '---') {
      return false
    }

    let line = 0
    let isFrontMatterClosed = false
    while (line < state.lineMax) {
      line++
      if (state.src.substring(state.bMarks[line], state.eMarks[line]) === '---') {
        isFrontMatterClosed = true
        break
      }
    }

    state.line = line + 1

    return isFrontMatterClosed
  }

  md.block.ruler.before('table', 'frontmatter', frontmatter, {
    alt: [ 'paragraph', 'reference', 'blockquote', 'list' ]
  })
}
```
The the modification above, the front-matter will be displayed out if it's invalid and will be hide only when it's valid.